smithy-swift-codegen/src/test/kotlin/serde/xml/UnionEncodeXMLGenerationTests.kt

```diff
@@ -65,93 +65,78 @@ class UnionEncodeXMLGenerationTests {
                             }
-                        } else {
-                            self = .mapvalue([:])
-                            return
-                        }
-                    } else {
-                        //No-op
-                    }
-                    if containerValues.contains(.stringlist) {
-                        struct KeyVal0{struct member{}}
-                        let stringlistWrappedContainer = containerValues.nestedContainerNonThrowable(keyedBy: CollectionMemberCodingKey<KeyVal0.member>.CodingKeys.self, forKey: .stringlist)
-                        if let stringlistWrappedContainer = stringlistWrappedContainer {
-                            let stringlistContainer = try stringlistWrappedContainer.decodeIfPresent([Swift.String].self, forKey: .member)
-                            var stringlistBuffer:[Swift.String]? = nil
-                            if let stringlistContainer = stringlistContainer {
-                                stringlistBuffer = [Swift.String]()
-                                for stringContainer0 in stringlistContainer {
-                                    stringlistBuffer?.append(stringContainer0)
+                        case .stringlist:
+                            if containerValues.contains(.stringlist) {
+                                struct KeyVal0{struct member{}}
+                                let stringlistWrappedContainer = containerValues.nestedContainerNonThrowable(keyedBy: CollectionMemberCodingKey<KeyVal0.member>.CodingKeys.self, forKey: .stringlist)
+                                if let stringlistWrappedContainer = stringlistWrappedContainer {
+                                    let stringlistContainer = try stringlistWrappedContainer.decodeIfPresent([Swift.String].self, forKey: .member)
+                                    var stringlistBuffer:[Swift.String]? = nil
+                                    if let stringlistContainer = stringlistContainer {
+                                        stringlistBuffer = [Swift.String]()
+                                        for stringContainer0 in stringlistContainer {
+                                            stringlistBuffer?.append(stringContainer0)
+                                        }
+                                    }
+                                    self = .stringlist(stringlistBuffer)
+                                } else {
+                                    self = .stringlist([])
                                 }
+                            } else {
+                                //No-op
                             }
-                            if let stringlist = stringlistBuffer {
-                                self = .stringlist(stringlist)
-                                return
+                        case .timestampvalue:
+                            let timestampvalueDecoded = try containerValues.decodeIfPresent(Swift.String.self, forKey: .timestampvalue)
```

Comment:
per smithy team, union associated values should never be null so if we get to this piont in the switch statement I think we should use decode not decodeIfPresent and skip the null check. so I think we can simplify this code further actually. less code  == smaller binary :) @wooj2

Reply:
handled offline, resolving.